src/elements/common/annotator-context/useAnnotatorEvents.ts

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { EventEmitter } from 'events';
+import { AnnotationActionEvent, Status } from './types';
+
+export interface UseAnnotatorEventsProps {
+    eventEmitter: EventEmitter;
+    onAnnotationDeleteEnd?: (annotationId: string) => void;
+    onAnnotationDeleteStart?: (annotationId: string) => void;
+    onAnnotationUpdateEnd?: (annotation: Object) => void;
+    onAnnotationUpdateStart?: (annotation: Object) => void;
+    onSidebarAnnotationSelected?: (annotationId: string) => void;
+}
+
+const emitUpdateAnnotationEvent = (eventEmitter: EventEmitter, annotation: Object, status: Status) => {
+    const actionEvent: AnnotationActionEvent = {
+        annotation,
+        meta: { status },
+    };
+    eventEmitter.emit('annotations_update', actionEvent);
+};
+
+function useAnnotatorEvents({
+    eventEmitter,
+    onAnnotationDeleteEnd,
+    onAnnotationDeleteStart,
+    onAnnotationUpdateEnd,
+    onAnnotationUpdateStart,
+    onSidebarAnnotationSelected,
+}: UseAnnotatorEventsProps) {
+    const emitAnnotationActiveChangeEvent = (annotationId: string | null, fileVersionId: string) => {
+        eventEmitter.emit('annotations_active_change', { annotationId, fileVersionId });
+    };
+
+    const emitUpdateAnnotationStartEvent = (annotation: Object) => {
+        emitUpdateAnnotationEvent(eventEmitter, annotation, Status.PENDING);
+    };
+    const emitUpdateAnnotationEndEvent = (annotation: Object) => {
+        emitUpdateAnnotationEvent(eventEmitter, annotation, Status.SUCCESS);
+    };
+
+    const annotationSidebarSelectedListener = (annotationId: string) => {
+        if (onSidebarAnnotationSelected) {
```

Comment:
Can we simplify this logic by setting these to `noop` functions? That way, we can pass them directly to the `addListener` calls.